### PR TITLE
Adds demo for `paymentHistory` method of `XRPClient` in Java

### PR DESCRIPTION
**Pull request by amiecorso**
## High Level Overview of Change
paymentHistory is the newest addition to the Xpring SDK XRPClient and needs to be reflected in the demos. 
This PR also updates the Java demos to reflect recent breaking changes to the SDK: `XpringException` recently became component-specific `IlpException` and `XRPException`.  Also, client constructors now take an optional parameter of enum type `XRPLNetwork`.

### Context of Change
`paymentHistory` returns an array of `XRPTransactions`.  These objects have no overloaded to-string or printing behavior.  While the printed objects aren't hideous, they could be more human-readable.  Future PRs will implement this behavior in the `XRPTransaction` class and others.

### Type of Change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

## Before / After
Java demos were broken, but now they compile.
`paymentHistory` method now reflected in demos.

## Test Plan
Circle

## Future Tasks

To-string and printing overloads for protobuf wrapper classes in SDKs.

### CODE REVIEW

**Review by 0xCLARITY — APPROVED** [review]

File: .gitignore [hunk 1]
```diff
@@ -6,7 +6,7 @@ java/.idea
 # Edit at https://www.gitignore.io/?templates=osx,node,java,xcode,swift,intellij,carthage
 
 ### Carthage ###
-# Carthage
+Carthage
```
Comment (keefertaylor) [inline .gitignore:9]: TY

**Comment by keefertaylor** [comment]
@nkramer44 as FYI that we uptook the breaking ILP changes.

File: java/src/main/java/io/xpring/demo/XrpDemo.java [hunk 2]
```diff
@@ -21,23 +24,34 @@ public static void main(String[] args) throws XpringException {
...
+        // Check status of the payment
         System.out.println("Hash for transaction:\n" + hash + "\n");
 
         TransactionStatus status = xrpClient.getPaymentStatus( hash);
         System.out.println("Result for transaction is:\n" + status + "\n");
+
+        // Retrieve full payment history for account
+        System.out.println("Payment history for account " + wallet.getAddress() +":\n");
+        List<XRPTransaction> paymentHistory = xrpClient.paymentHistory(wallet.getAddress());
+        for (XRPTransaction transaction : paymentHistory) {
```
Comment (keefertaylor) [inline java/src/main/java/io/xpring/demo/XrpDemo.java:53]: This demo may be slightly more slick if we limit this to the 5 most recent transactions (and obviously print that we're doing that)

**Review by keefertaylor — APPROVED** [review]
Seems reasonable to me. I'd limit the number of transactions to make the demo more readable on a presentation screen, but open to pushback there.

**Review by keefertaylor — COMMENTED** [review]
Undo approval; this doesn't build :)

**Comment by keefertaylor** [comment]
I think this isn't building because we need to upgrade the version of Xpring4j (which is defined here: 
https://github.com/xpring-eng/Xpring-SDK-Demo/blob/master/java/pom.xml#L19) 

Unfortunately, x4j needs a new release cut. I can kick one of those off now.

**Comment by keefertaylor** [comment]
Release cut here: https://github.com/xpring-eng/xpring4j/pull/194

**Review by 0xCLARITY — APPROVED** [review]
LGTM. Good job adding comments in the `XRPDemo.java` file!

**Pull request merged**